contrib/automation_tests/test_cases/connection_window.py

```diff
@@ -95,13 +96,93 @@ def _execute(self):
+        # wait until loading is done
+        wait_for_condition(lambda: project_combo_box.is_enabled() is True, 25)
+        project_combo_box.click_input()
+
+        default_project = self.find_combo_box_item(text=project_name)
+        logging.info('Expanded project combo box and found entry ' + project_name)
+        default_project.click_input()
+        # After project changing, loading takes place. Wait until the overlay is hidden
+        wait_for_condition(
+            lambda: self.find_control('Group', 'InstanceListOverlay', raise_on_failure=False) is
+            None, 100)
+        logging.info('Successfully selected project ' + project_name +
+                     'and waited until loading is done')
+
+
+class SelectNextProject(E2ETestCase):
+    """
+    Expand the project selection combo box and select the next project by pressing the down arrow
+    and enter. Also waits appropriately when loading takes place.
+    """
+
+    def _execute(self):
+        logging.info('Select next project')
+        project_combo_box = self.find_control(name='ProjectSelectionComboBox')
+        # wait until loading is done
+        wait_for_condition(lambda: project_combo_box.is_enabled() is True, 25)
+        project_combo_box.click_input()
+        logging.info('Found and opened project combo box, sending down arrow and enter')
+        send_keys('{DOWN}{ENTER}')
+        # After project changing, loading takes place. Wait until the overlay is hidden
+        wait_for_condition(
+            lambda: self.find_control('Group', 'InstanceListOverlay', raise_on_failure=False) is
+            None, 100)
+        logging.info('Successfully selected next project')
+
+
+class TestAllInstancesCheckbox(E2ETestCase):
+    """
+    This test expects that the "All Instances" check box is not checked at the start. The Test
+    clicks the check box and expects that the number of instances stays the same or increases.
+    Afterwards the checkbox is clicked again to reset the state.
+    """
+
+    def _execute(self):
+        logging.info('Starting "All Instances" checkbox test.')
+        # First wait until all loading is done.
+        wait_for_condition(
+            lambda: self.find_control('Group', 'InstanceListOverlay', raise_on_failure=False) is
+            None, 100)
+        check_box = self.find_control('CheckBox', 'AllInstancesCheckBox')
+        logging.info('Found checkbox')
+
+        instance_count = get_number_of_instances_in_list(self)
+
+        check_box.click_input()
+        logging.info('Clicked All Instances check box, waiting until loading is done')
+        wait_for_condition(
+            lambda: self.find_control('Group', 'InstanceListOverlay', raise_on_failure=False) is
+            None, 100)
+        self.expect_true(instance_count <= get_number_of_instances_in_list(self),
+                         'Instance list contains at least same amount of instances as before.')
```

Comment:
This is a very weak condition to check... I don't know how the test is setup, would it be possible to make sure there is at least one instance of another user to make this comparison "<"?

Reply:
No, that is not possible. In theory, of course one could write test runner code to do that, but its a lot of work